### PR TITLE
NT 140 intro Instruction cutscene

### PR DESCRIPTION
**Pull request by PatrickSlag**

### CODE REVIEW

File: Assets/Scripts/GameManager.cs [hunk 1]
```diff
@@ -159,4 +169,80 @@ private void FillInventory()
...
+        _gameState = GameState.MovingIsland;
+    }
+
+    private void StartInstruction()
+    {
+        _dialoguePresenter.StartGeneralInstruction();
+        softClouds.SetActive(true);
+    }
+
+    /*private void OnCutSceneEnd(PlayableDirector director)
```
Comment (VarioCode) [inline Assets/Scripts/GameManager.cs]: Commented out logic can be removed

File: Assets/Scripts/GameManager.cs [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
+using System.Runtime.CompilerServices;
+using Cinemachine;
 using Enums;
 using Models;
+using Presenters;
 using UnityEngine;
+using UnityEngine.Playables;
+using UnityEngine.UI;
 
 public class GameManager : MonoBehaviour
```
Comment (VarioCode) [inline Assets/Scripts/GameManager.cs:11]: The game manager should only be used for managing state...why was so much logic added here?

File: Assets/Scripts/GameManager.cs [hunk 3]
```diff
@@ -159,4 +169,80 @@ private void FillInventory()
...
+    {
+        if(_gameState == GameState.MovingIsland)
+        {
+            island.transform.position = Vector3.MoveTowards(island.transform.position, endMarker.position, speed);
+            if (island.transform.position == endMarker.position && _stormParticleSystem.isStopped)
+            {
+                startCamera.SetActive(false);
+                playableDirector.gameObject.SetActive(true);
+                playableDirector.Play();
+                //playableDirector.stopped += OnCutSceneEnd;
```
Comment (VarioCode) [inline Assets/Scripts/GameManager.cs]: Commented out logic can be removed

File: Assets/Scripts/Presenters/DialoguePresenter.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+using UnityEngine.UI;
+
+namespace Presenters
+{
+    public class DialoguePresenter : MonoBehaviour
+    {
+        [SerializeField] private GameObject dialogueBox;
+        [SerializeField] private TMP_Text dialogueText;
+        [SerializeField] private Button nextDialogueButton;
+        //[SerializeField] private Button DebugDialogueButton;
```
Comment (VarioCode) [inline Assets/Scripts/Presenters/DialoguePresenter.cs]: Commented out logic can be removed

File: Assets/Scripts/Presenters/DialoguePresenter.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            };
+            
+            QueMessages(testMessages);
+            ShowNextMessage();
+        }
+
+        private void Start()
+        {
+            nextDialogueButton.onClick.AddListener(ShowNextMessage);
+            //DebugDialogueButton.onClick.AddListener(DebugStartDialogue);
```
Comment (VarioCode) [inline Assets/Scripts/Presenters/DialoguePresenter.cs]: Commented logic please remove

File: Assets/Scripts/Presenters/DialoguePresenter.cs [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+        public void StartGeneralInstruction()
+        {
+            QueMessages(_dialogueGenerationService.SplitTextToInstructionMessages(GetInstruction("general")));
+            ShowNextMessage();
+        }
+        
+        /*//Debug UI
```
Comment (VarioCode) [inline Assets/Scripts/Presenters/DialoguePresenter.cs]: Can be deleted

**Review by VarioCode — CHANGES_REQUESTED** [review]
I made a few comment either fix the issues or create a TD story for them. If you choose the 2nd option, please comment it under this message so I can go over it and see if I miss anything.

File: Assets/Scripts/GameManager.cs (hunk 2 above)
Comment (PatrickSlag) [reply Assets/Scripts/GameManager.cs:11]: https://038games.atlassian.net/browse/NT-217

Comment (VarioCode) [reply Assets/Scripts/GameManager.cs:11]: Alr gtg

**Review by VarioCode — APPROVED** [review]

**Review by jortdr — APPROVED** [review]

**Pull request merged**